c10/util/Float8_fnuz_cvt.h

```diff
@@ -33,6 +33,8 @@ inline C10_HOST_DEVICE float fp8_fnuz_to_fp32_value(uint8_t x) {
     // guaranteed mantissa!=0 since cases 0x0 and 0x80 are handled above
 #if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
     uint32_t renorm_shift = __clz(mantissa);
+#elif defined(__SYCL_DEVICE_ONLY__)
+    uint32_t renorm_shift = __builtin_clz(mantissa);
```

Comment:
@ratnampa please upstream fix to this, and I'll rebase

Reply:
This line is totally same with line #43, why we do not reuse it?

Reply:
Reason is, we need to use this path for SYCL device code before `_MSC_VER` path for host code.

Reply:
Better as below?

```suggestion
#elif defined(_MSC_VER) and not defined(__SYCL_DEVICE_ONLY__)
    unsigned long nonsign_bsr;
    _BitScanReverse(&nonsign_bsr, (unsigned long)mantissa);
    uint32_t renorm_shift = (uint32_t)nonsign_bsr ^ 31;
#else
    uint32_t renorm_shift = __builtin_clz(mantissa);
```

Reply:
Yes, thanks.